common:
    lidar_topic:  "/os_cloud_node/points"
    imu_topic:  "/vn200/imu"
    point_filter_num: 2
    gravity_acc: [ 0.0, 0.0, 9.8065]

lidar_parameter:
    lidar_type: 3                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    N_SCANS: 64
    SCAN_RATE: 10                # only need to be set for velodyne, unit: Hz,
    time_unit: 3            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 1
    fov_degree:    360
    det_range:     150.0

imu_parameter:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    time_diff_enable: false

extrinsic_parameter:
    extrinsic_enable:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_t: [-0.04894521120476690,  -0.03126929060277820, -0.01755515794190739] # transforms a point from lidar into imu frame
    extrinsic_R: [ 0.999913504074182800, -0.01116636551106650, -0.0069495792219705765,
                  -0.011356389542500523, -0.99954530068657550, -0.0279324952688129200,
                  -0.006634514801262056,  0.02800900135057546, -0.9995856536869142000]

odometry_options:
    voxel_size: 0.5 # The voxel size for the grid sampling of the new frame (before keypoints extraction)
    sample_voxel_size: 1.5 # The size of a voxel for the selection of `keypoints` by grid sampling
    max_distance: 2000.0 # The threshold of the distance to suppress voxels from the map
    max_num_points_in_voxel: 20 # The maximum number of points per voxel of the map
    method_system_init: STATIC_INIT # LIO system initialization [MOTION_INIT, STATIC_INIT]
    init_num_frames: 20
    robust_registration: false
    min_distance_points: 0.15
    distance_error_threshold: 100.0 # The motion of the sensor between two frames which is considered erroneous (stops the odometry)
    motion_compensation: CONSTANT_VELOCITY # The profile of the motion compensation (NONE, CONSTANT_VELOCITY, ITERATIVE, CONTINUOUS)
    initialization: INIT_IMU # [INIT_CONSTANT_VELOCITY, INIT_IMU]

icp_options:
    size_voxel_map: 1.0 # The voxel size of in the voxel map
    num_iters_icp: 5 # The number of iterations of the ICP
    min_number_neighbors: 20 # The minimum number of neighbor points to define a valid neighborhood
    voxel_neighborhood: 1
    max_number_neighbors: 20
    max_dist_to_plane_ct_icp: 0.3
    threshold_orientation_norm: 0.1 # Threshold on orientation changes (in degrees) for early termination of the ICP
    threshold_translation_norm: 0.01 # Threshold on distance changes (in m) for early termination of the ICP
    debug_print: false
    point_to_plane_with_distortion: true
    distance: POINT_TO_PLANE # Options: [CT_POINT_TO_PLANE, POINT_TO_PLANE]
    num_closest_neighbors: 1
    beta_location_consistency: 1.0
    beta_orientation_consistency: 1.0
    beta_constant_velocity: 1.0 # Trajectory Regularisation Weight to constrain the optimization
    beta_small_velocity: 0.00
    solver: LIO # The type of SOLVER used in [LIO, LIDAR]
    min_num_residuals: 200 # The minimum number of residuals for a valid ICP problem
    max_num_residuals: 600 # The maximum number of residuals considered (if more keypoints exist, residuals are randomly sampled)

    loss_function: HUBER # Options: [CAUCHY, STANDARD, HUBER, TOLERANT, TRUNCATED]
    ls_max_num_iters: 5 # The number of steps performed by ceres for each iteration of the ICP
    ls_num_threads: 16 # The number of threads to build and solve the least square system
    ls_sigma: 0.5
    ls_tolerant_min_threshold: 0.05

